metron-platform/metron-parsing/metron-parsing-storm/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -326,6 +326,10 @@ protected void handleError(String sensorType, byte[] originalMessage, Tuple tupl
             .withThrowable(ex)
             .withSensorType(Collections.singleton(sensorType))
             .addRawMessage(originalMessage);
+    handleError(collector, error);
+  }
+
+  protected void handleError(OutputCollector collector, MetronError error) {
```

Comment:
Hey, I like this pattern you used here of pushing the external dep into a function. Any future refactoring on code that does this becomes a bit easier. Also:

- Easier to read
- Gives a more consistent abstraction level within the main method
- Encourages good composition patterns, which is great architecturally as well as for unit testing